cmd/csi-resizer/main.go

```diff
@@ -235,13 +237,46 @@ func main() {
 			workqueue.NewTypedItemExponentialFailureRateLimiter[string](*retryIntervalStart, *retryIntervalMax))
 	}
 
+	// handle SIGTERM and SIGINT by cancelling the context.
+	var (
+		terminate       func()          // called when all controllers are finished
+		controllerCtx   context.Context // shuts down all controllers on a signal
+		shutdownHandler <-chan struct{} // called when the signal is received
+	)
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.ReleaseLeaderElectionOnExit) {
+		ctx, terminate = context.WithCancel(ctx) // shuts down the whole process, incl. leader election
+		var cancelControllerCtx context.CancelFunc
+		controllerCtx, cancelControllerCtx = context.WithCancel(ctx)
+		shutdownHandler = server.SetupSignalHandler()
+
+		defer terminate()
+
+		go func() {
+			defer cancelControllerCtx()
+			<-shutdownHandler
+			klog.Info("Received SIGTERM or SIGINT signal, shutting down controller.")
+		}()
+	}
+
 	run := func(ctx context.Context) {
 		informerFactory.Start(wait.NeverStop)
```

Comment:
Should `informerFactory` be shutdown when interrupt signal is received? It may have been an oversight to use `wait.NeverStop` here. I do not like the idea of processing informer events when things are getting shutdown.